image-service/src/fileAdapter.ts

```diff
@@ -0,0 +1,26 @@
+import fs from "fs";
+import logger from "./logger";
+import {IStorageAdapter} from "./storage";
+
+export default class FileAdapter implements IStorageAdapter {
+    constructor(private source: string) {
+    }
+
+    public async save(oldPath: string, fileName: string): Promise<string> {
+        const newPath = this.getNewPath(fileName);
+        try {
+            // @ts-ignore
+            await fs.rename(oldPath, newPath, (e) => {
```

Comment:
fs.rename does not return promise, you need promisified version https://nodejs.org/dist/latest-v10.x/docs/api/util.html#util_util_promisify_original

Reply:
fixed